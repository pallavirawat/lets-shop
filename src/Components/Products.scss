
.Products{
  display: flex;
  align-items: center;
    justify-content: center;
}

.Card{
  margin-right: 75px;
  width: 18rem;
}