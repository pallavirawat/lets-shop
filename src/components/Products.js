import * as PropTypes from "prop-types";
import React, {useState} from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import {bootstapVariant, currencySymbol, productlist} from "../data/constants";
import './Products.scss'
import {changePlayerStatus, getRandomDecision, restartShopping} from "../engineDynamics/Engine";
import StageResult from "./StageResult";

function Products(props) {
    const [stageDecision, setStageDecision] = useState()
    const [resultShow, setResultShow] = useState(false);

    const handleBuyNow = () => {
        const decision = getRandomDecision()
        console.log(decision)
        setStageDecision(decision)
        setResultShow(true)
    }

    return <div>
        <div className="Products">
            {
                productlist.map((product, idx) => (
                        <Card className="Card" key={idx} bg={bootstapVariant.WARNING} border={bootstapVariant.LIGHT}>
                            <Card.Img variant="top" src={product.image}/>
                            <Card.Body>
                                <Card.Title>{product.name}</Card.Title>
                                <Card.Text>
                                    {`${product.cost} ${currencySymbol}`}
                                </Card.Text>
                                <Button variant={bootstapVariant.INFO} onClick={handleBuyNow}>Buy Now</Button>
                            </Card.Body>
                        </Card>
                    )
                )
            }

        </div>
        ;
        <StageResult decision={stageDecision} show={resultShow} setShow={setResultShow}
                     onProceed={() => {
                         changePlayerStatus(stageDecision,
                             props.creditLimit, props.setCreditLimit,
                             props.deathCount, props.setDeathCount,
                             props.stage, props.setStage)
                         setResultShow(false)
                     }}
                     handleRestart={() => {
                         restartShopping(props.setCreditLimit, props.setDeathCount, props.setStage)
                         setResultShow(false)
                     }}
        />
    </div>
}

Products.propTypes = {
    deathCount: PropTypes.number.isRequired,
    creditLimit: PropTypes.number.isRequired,
    stage: PropTypes.number.isRequired,
    setDeathCount: PropTypes.func.isRequired,
    setCreditLimit: PropTypes.func.isRequired,
    setStage: PropTypes.func.isRequired,
};

Products.defaultProps = {}
export default Products