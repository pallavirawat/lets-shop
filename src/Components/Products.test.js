import {render} from "@testing-library/react";
import React from "react";
import Products from "./Products";

describe('Products', function () {
    it('should render', function () {
        expect(render(<Products deathCount={3}/>)).toMatchSnapshot();
    });
});