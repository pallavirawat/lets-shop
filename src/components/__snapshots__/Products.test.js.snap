// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="Products"
        >
          <div
            class="Card card bg-warning border-light"
          >
            <img
              class="card-img-top"
              src="https://assets.myntassets.com/h_720,q_90,w_540/v1/assets/images/11375892/2020/8/14/75a40b63-cc29-4dcb-b184-2d181188a0701597410903057BaggitBrownSolidSlingBag1.jpg"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                Random Bag
              </div>
              <p
                class="card-text"
              >
                35 $
              </p>
              <button
                class="btn btn-info"
                type="button"
              >
                Buy Now
              </button>
            </div>
          </div>
          <div
            class="Card card bg-warning border-light"
          >
            <img
              class="card-img-top"
              src="https://assets.myntassets.com/h_720,q_90,w_540/v1/assets/images/11334994/2020/5/21/3d167b8c-e00e-4a53-9f3f-b7e4e2f2a4481590080005795PumaMenBlackPacerStyxIDPSneakers1.jpg"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                Puma Shoe
              </div>
              <p
                class="card-text"
              >
                50 $
              </p>
              <button
                class="btn btn-info"
                type="button"
              >
                Buy Now
              </button>
            </div>
          </div>
          <div
            class="Card card bg-warning border-light"
          >
            <img
              class="card-img-top"
              src="https://assets.myntassets.com/h_720,q_90,w_540/v1/assets/images/1392352/2016/10/4/11475579108886-Fossil-Men-Navy-Dial-Watch-ME1138-3141475579108667-1.jpg"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                Fossil Watch
              </div>
              <p
                class="card-text"
              >
                40 $
              </p>
              <button
                class="btn btn-info"
                type="button"
              >
                Buy Now
              </button>
            </div>
          </div>
        </div>
        ;
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="Products"
      >
        <div
          class="Card card bg-warning border-light"
        >
          <img
            class="card-img-top"
            src="https://assets.myntassets.com/h_720,q_90,w_540/v1/assets/images/11375892/2020/8/14/75a40b63-cc29-4dcb-b184-2d181188a0701597410903057BaggitBrownSolidSlingBag1.jpg"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Random Bag
            </div>
            <p
              class="card-text"
            >
              35 $
            </p>
            <button
              class="btn btn-info"
              type="button"
            >
              Buy Now
            </button>
          </div>
        </div>
        <div
          class="Card card bg-warning border-light"
        >
          <img
            class="card-img-top"
            src="https://assets.myntassets.com/h_720,q_90,w_540/v1/assets/images/11334994/2020/5/21/3d167b8c-e00e-4a53-9f3f-b7e4e2f2a4481590080005795PumaMenBlackPacerStyxIDPSneakers1.jpg"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Puma Shoe
            </div>
            <p
              class="card-text"
            >
              50 $
            </p>
            <button
              class="btn btn-info"
              type="button"
            >
              Buy Now
            </button>
          </div>
        </div>
        <div
          class="Card card bg-warning border-light"
        >
          <img
            class="card-img-top"
            src="https://assets.myntassets.com/h_720,q_90,w_540/v1/assets/images/1392352/2016/10/4/11475579108886-Fossil-Men-Navy-Dial-Watch-ME1138-3141475579108667-1.jpg"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Fossil Watch
            </div>
            <p
              class="card-text"
            >
              40 $
            </p>
            <button
              class="btn btn-info"
              type="button"
            >
              Buy Now
            </button>
          </div>
        </div>
      </div>
      ;
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
